src/concurrency/sync.rs

```diff
@@ -731,9 +731,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     let futex = this.machine.sync.futexes.get_mut(&addr).unwrap();
                     futex.waiters.retain(|waiter| waiter.thread != thread);
                     // Set errno and write return value.
-                    this.set_last_error(errno_timeout)?;
-                    this.write_scalar(retval_timeout, &dest)?;
-                    interp_ok(())
+                    this.set_last_error_and_return(errno_timeout, &dest)
```

Comment:
Checked the types in futex_wait and made this change - changing the type of errno_timeout broke some other functions downstream: a call to `eval_windows` in the Windows shims passed to futex_wait broke, and in concurrency/thread.rs, a `self.$name.visit_provenance(visit)` isn't implemented for IoErrors. 

I could probably add a `WindowsError` variant to the IoError enum to fix the shims error. I could also fill out more of the Windows error table while I'm at it. Less sure about the architecture of the provenance types, to be honest.

Reply:
Okay let's defer that to a future PR then.

(The right thing would indeed be to add a `WindowsError` variant, and to implement the `VisitProvenance` trait for `IoError`.)